### PR TITLE
hashes: Remove `utils` module

### PR DESCRIPTION
**Pull request by tcharding**
We don't want dump-all module names. Remove `utils` in favour of `macros`, while we are at it roll the `serde_maros` code into it as well and remove that module.

Fix: #2665

### CODE REVIEW

**Comment by Kixunil** [comment]
The API break is real, we need to deprecate.

**Comment by tcharding** [comment]
I didn't test the deprecation, I remember it not working when I've tried this in the past but it will be interesting to see what the semver bot thinks of it.

**Comment by tcharding** [comment]
Cool, the bot likes my PR.

File: hashes/src/lib.rs [hunk 1]
```diff
@@ -128,6 +126,17 @@ pub mod sha512;
 pub mod sha512_256;
 pub mod siphash24;
 
+#[deprecated(since = "TBD", note = "use crate::macros instead")]
```
Comment (Kixunil) [inline hashes/src/lib.rs:129]: Yes, this is correct.

**Review by Kixunil — DISMISSED** [review]
ACK a959c95afb70fa3722e67e62e664c1ee823bcc35

**Comment by tcharding** [comment]
Rebased and improved commit logs, no other changes. Note the range diff has 

```diff
   - #[cfg(feature = "bitcoin-io")]
    - mod impls;
    + extern crate schemars;
    + 
```

And that led me to #3374

**Comment by apoelstra** [comment]
(a) needs rebase again, apparently, and (b) ```diff should work. But I think the stuff you quoted is just too small and headerless to be recognizable as a diff.

**Comment by tcharding** [comment]
#3354 conflicts with this because of the `SerdeHash` stuff, I rekon that one should go in first. Converting to draft.

**Review by apoelstra — APPROVED** [review]
ACK 092d4abb443bb1d9ae8e47d62c002f74c6b483c8; successfully ran local tests

**Comment by tcharding** [comment]
Merge please.

**Review by apoelstra — DISMISSED** [review]
ACK 7589d90b73448132237f9516291a353f65fba3f0; successfully ran local tests; but needs rebase

**Comment by tcharding** [comment]
Rebased, no other changes.

**Review by apoelstra — APPROVED** [review]
ACK 5a736edaaac5e13166dc907c8608b84132c65096; successfully ran local tests

**Pull request merged**